### PR TITLE
Apply jdt clean ups

### PR DESCRIPTION
**Pull request by HannesWell**
Apply JDT clean ups

- "Use diamond operator"
- "Use try-with-resource"
- "Use try with multi-catch"

to Equinox-sources and ignore embedded felix and osgi sources.

The last clean up required manual removal of now unused Exception type imports but that problem has been reported to JDT.

### CODE REVIEW

**Comment by HannesWell** [comment]
The test failures is only for the GH workflow. Jenkins and local execution of that test pass.
Merging.

File: bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/FrameworkExtensionInstaller.java [hunk 1]
```diff
@@ -80,8 +80,6 @@ private static Method findMethod(Class<?> clazz, String name, Class<?>[] args, C
 				result.setAccessible(true);
 			}
 			return result;
-		} catch (SecurityException e) {
-			// if we do not have the permissions then we will not find the method
```
Comment (tjwatson) [inline bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/FrameworkExtensionInstaller.java:84]: Why did you remove this?  I believe this will result in blowing up `<clinit>` if the framework does not have permission to reflect.

Comment (tjwatson) [reply bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/storage/FrameworkExtensionInstaller.java:84]: Nevermind.  SecurityException is a RuntimeException!
